### PR TITLE
AToTB: Full Set of Hints

### PR DESCRIPTION
**Pull request by sigurdfdragon**
Uses an expanded form of the hint macro from AOI and adds hints for each objective in the campaign.

S01-04 are separate commits in case changes are needed, I'll squash them later.

I'll leave this up at least a week for comments or suggestions on the hints.

### CODE REVIEW

File: data/campaigns/Two_Brothers/scenarios/02_The_Chase.cfg [hunk 1]
```diff
@@ -299,6 +299,45 @@ Besides... I want my brother back."
...
+                    [/have_unit]
+                    [not]
+                        [variable]
+                            name=found_kidnappers
+                            boolean_equals=yes
+                        [/variable]
+                    [/not]
+                [/show_if]
+            )}
+            {HINT ( _ "Use horsemen or footpads to quickly explore the the shrouded woods to locate the kidnappers.") (
```
Comment (sevu) [inline data/campaigns/Two_Brothers/scenarios/02_The_Chase.cfg]: here's a "the" to much

File: data/campaigns/Two_Brothers/scenarios/03_Guarded_Castle.cfg [hunk 2]
```diff
@@ -323,6 +323,40 @@ Yet for some reason I fear these brothers more. If Mordak were here it would be
...
+            {HINT ( _ "When facing an unknown situation, take into account details from story and dialog to inform your strategy.") (
+                [show_if]
+                    [not]
+                        [have_unit]
+                            id=Baran
+                        [/have_unit]
+                    [/not]
+                [/show_if]
+            )}
+            {HINT ( _ "Lure the dark sorcerer off of his keep by giving him a shot at a melee unit, then counter-attack in force.") (
```
Comment (jostephd) [inline data/campaigns/Two_Brothers/scenarios/03_Guarded_Castle.cfg]: This hint just tells the player how to exploit an AI shortsightedness, doesn't it?  I.e., if a human player controlled the dark sorcerer, this strategy wouldn't work.

(I'd _love_ to be able to play the mainline campaigns with humans controlling the enemy sides, by the way.)

Comment (GregoryLundberg) [reply data/campaigns/Two_Brothers/scenarios/03_Guarded_Castle.cfg]: The idea of these hints is to introduce new players to the game. Yes, it's an AI weakness, and no human would fall for the trick. So what? We want the new player to succeed. If that means showing them the "AI" aint that "I" then so be it.,

Comment (jostephd) [reply data/campaigns/Two_Brothers/scenarios/03_Guarded_Castle.cfg]: @GregoryLundberg I think we want new players to succeed even against strong opponents.  How about rephrasing the advice to be more generic, e.g., "The dark sorcerer is a strong unit; attack him with multiple units at once and try to force him off of his keep."?

**Pull request merged**